### PR TITLE
JavaScript samples

### PR DESCRIPTION
**Pull request by fraguada**
I've been creating some samples that target different Javascript frameworks / runtimes. The samples here include running Rhino inside Node.js, Electron.js, and CefSharp. I'd like to have someone test that they can follow the instructions and run the samples on Windows. Many of these, except for probably Sample-5, should work on macOS (if Rhino WIP for mac OS supported Rhino.Inside) as well, but I will work on that (lobbying for it) after / if this gets merged.

### CODE REVIEW

File: JavaScript/Sample-1/insideNode/insideNode.csproj [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="RhinoCommon">
+      <HintPath>..\..\..\..\..\..\Program Files\Rhino WIP\System\RhinoCommon.dll</HintPath>
```
Comment (pearswj) [inline JavaScript/Sample-1/insideNode/insideNode.csproj]:
```suggestion
      <HintPath>C:\Program Files\Rhino WIP\System\RhinoCommon.dll</HintPath>
```

File: JavaScript/Sample-2/insideNode/InsideNode.csproj [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="RhinoCommon">
+      <HintPath>..\..\..\..\..\..\Program Files\Rhino WIP\System\RhinoCommon.dll</HintPath>
```
Comment (pearswj) [inline JavaScript/Sample-2/insideNode/InsideNode.csproj]:
```suggestion
      <HintPath>C:\Program Files\Rhino WIP\System\RhinoCommon.dll</HintPath>
```

File: JavaScript/Sample-3/InsideElectron/InsideElectron.csproj [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    <PackageReference Include="Microsoft.DotNet.InternalAbstractions" Version="1.0.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="2.0.0" />
+  </ItemGroup>
+  
+  <ItemGroup>
+    <Reference Include="EdgeJs">
+      <HintPath>..\InsideElectronApp\node_modules\electron-edge-js\lib\bootstrap\bin\Release\netcoreapp1.1\EdgeJs.dll</HintPath>
+    </Reference>
+    <Reference Include="RhinoCommon">
+      <HintPath>..\..\..\..\..\..\..\..\Program Files\Rhino WIP\System\RhinoCommon.dll</HintPath>
```
Comment (pearswj) [inline JavaScript/Sample-3/InsideElectron/InsideElectron.csproj]:
```suggestion
      <HintPath>C:\Program Files\Rhino WIP\System\RhinoCommon.dll</HintPath>
```

File: JavaScript/Sample-4/InsideElectron/InsideElectron.csproj [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="GH_IO">
+      <HintPath>..\..\..\..\..\..\Program Files\Rhino WIP\Plug-ins\Grasshopper\GH_IO.dll</HintPath>
```
Comment (pearswj) [inline JavaScript/Sample-4/InsideElectron/InsideElectron.csproj]:
```suggestion
      <HintPath>C:\Program Files\Rhino WIP\Plug-ins\Grasshopper\GH_IO.dll</HintPath>
```

File: JavaScript/Sample-4/InsideElectron/InsideElectron.csproj [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      <Private>False</Private>
+    </Reference>
+    <Reference Include="Grasshopper">
+      <HintPath>..\..\..\..\..\..\Program Files\Rhino WIP\Plug-ins\Grasshopper\Grasshopper.dll</HintPath>
```
Comment (pearswj) [inline JavaScript/Sample-4/InsideElectron/InsideElectron.csproj]:
```suggestion
      <HintPath>C:\Program Files\Rhino WIP\Plug-ins\Grasshopper\Grasshopper.dll</HintPath>
```

File: JavaScript/Sample-4/InsideElectron/InsideElectron.csproj [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      <Private>False</Private>
+    </Reference>
+    <Reference Include="RhinoCommon">
+      <HintPath>..\..\..\..\..\..\Program Files\Rhino WIP\System\RhinoCommon.dll</HintPath>
```
Comment (pearswj) [inline JavaScript/Sample-4/InsideElectron/InsideElectron.csproj]:
```suggestion
      <HintPath>C:\Program Files\Rhino WIP\System\RhinoCommon.dll</HintPath>
```

File: JavaScript/Sample-5/InsideCEF.WinForms/InsideCEF.WinForms.csproj [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    <DefineConstants>DEBUG;WinForms</DefineConstants>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|x64'">
+    <PlatformTarget>x64</PlatformTarget>
+    <DefineConstants>WinForms</DefineConstants>
+    <Prefer32Bit>false</Prefer32Bit>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="GH_IO">
+      <HintPath>..\..\..\..\..\..\Program Files\Rhino WIP\Plug-ins\Grasshopper\GH_IO.dll</HintPath>
```
Comment (pearswj) [inline JavaScript/Sample-5/InsideCEF.WinForms/InsideCEF.WinForms.csproj]:
```suggestion
      <HintPath>C:\Program Files\Rhino WIP\Plug-ins\Grasshopper\GH_IO.dll</HintPath>
```

File: JavaScript/Sample-5/InsideCEF.WinForms/InsideCEF.WinForms.csproj [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    </Reference>
+    <Reference Include="Grasshopper">
+      <HintPath>..\..\..\..\..\..\Program Files\Rhino WIP\Plug-ins\Grasshopper\Grasshopper.dll</HintPath>
```
Comment (pearswj) [inline JavaScript/Sample-5/InsideCEF.WinForms/InsideCEF.WinForms.csproj]:
```suggestion
      <HintPath>C:\Program Files\Rhino WIP\Plug-ins\Grasshopper\Grasshopper.dll</HintPath>
```

File: JavaScript/Sample-5/InsideCEF.WinForms/InsideCEF.WinForms.csproj [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+      <Private>False</Private>
+    </Reference>
+    <Reference Include="RhinoCommon">
+      <HintPath>..\..\..\..\..\..\Program Files\Rhino WIP\System\RhinoCommon.dll</HintPath>
```
Comment (pearswj) [inline JavaScript/Sample-5/InsideCEF.WinForms/InsideCEF.WinForms.csproj]:
```suggestion
      <HintPath>C:\Program Files\Rhino WIP\System\RhinoCommon.dll</HintPath>
```

**Comment by pearswj** [comment]
@fraguada I tried Sample 4 and found that it was necessary to navigate to `InsideElectronApp` and run `npm install` before building `InsideElectron.sln`. Also, Rhino seems to get stuck "_Downloading profile picture from Rhino account..._" – if I hit Cancel it continues as normal and I can create meshes in Grasshopper and preview them in Electron 🥳

**Comment by pearswj** [comment]
Unfortunately the technique of running `npm install` before compiling isn't working for me with Sample 3.

One more thing... Sample 5's default platform is AnyCPU but I had to switch to x64 to get it to build.

**Comment by fraguada** [comment]
>  I tried Sample 4 and found that it was necessary to navigate to InsideElectronApp and run npm install before building InsideElectron.sln. 

Ok.

> Also, Rhino seems to get stuck "Downloading profile picture from Rhino account..." – if I hit Cancel it continues as normal and I can create meshes in Grasshopper and preview them in Electron 🥳

Yeah, I also saw this and thought it was something to do with my connection. I'll check it out.

> One more thing... Sample 5's default platform is AnyCPU but I had to switch to x64 to get it to build.
It should have been set to x64 🤔. I'll check it out. Thanks.

**Comment by pearswj** [comment]
Sample 4 works for me now! :shipit:

**Comment by fraguada** [comment]
@pearswj I think I've resolved all previous issues you've reported:

- Sample-3: README should reflect correct instruction sequence.
- Sample-4: Issue with license still fetching profile pic/license. This should be fixed
- Sample-5: csproj should be set to x64

Still, Sample 4 should have the correct set up sequence (build csproj, then npm install). Sample 3 does depend on the contents of npm install, so the instructions were incorrect. Sample-3 instructions should be correct now.

Kindly check it out at your earliest convenience.

**Comment by pearswj** [comment]
@fraguada I was too keen 😅. Sample 5 loads x64 first time now and the Sample 3 readme looks good :+1:

**Pull request merged**